From c5666afc2b7562dd7e61a2cdb03c1752166d17ab Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Thu, 23 Aug 2018 09:29:32 +0300
Subject: [PATCH 4459/5242] MLK-19010: drm/bridge: nwl-dsi: Fix DSI long read

commit  8955ac3db80cbca56acb9783563a48130721d645 from
https://source.codeaurora.org/external/imx/linux-imx.git

On the long read routine, there was a bug when the driver was parsing
the value from RX_PAYLOAD register.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/nwl-dsi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/nwl-dsi.c b/drivers/gpu/drm/bridge/nwl-dsi.c
index 9ac8447..8cad5f4 100644
--- a/drivers/gpu/drm/bridge/nwl-dsi.c
+++ b/drivers/gpu/drm/bridge/nwl-dsi.c
@@ -695,7 +695,7 @@ static bool nwl_dsi_read_packet(struct nwl_mipi_dsi *dsi, u32 status)
 			payload[1] = (val >>  8) & 0xff;
 			++xfer->rx_len;
 			/* Fall through */
-		case 0:
+		case 1:
 			payload[0] = (val >>  0) & 0xff;
 			++xfer->rx_len;
 			break;
-- 
1.7.9.5

